### PR TITLE
backend/feat-182-added-special-loc-in-lib-and-serviceability-update

### PR DESCRIPTION
**Pull request by umangj-28**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Added special location table in lib
Servicability API checks for special Location

### Additional Changes

- [x] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Motivation and Context

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Backend/dev/migrations/dynamic-offer-driver-app/0127-updated-transporter-config-table.sql
Comment (RitikaHanish) [inline Backend/dev/migrations/dynamic-offer-driver-app/0127-updated-transporter-config-table.sql:1]: Why this change ?

File: Backend/dev/migrations/dynamic-offer-driver-app/0128-special-zone.sql
Comment (RitikaHanish) [inline Backend/dev/migrations/dynamic-offer-driver-app/0128-special-zone.sql:1]: Why this change is required ?

File: Backend/dev/migrations/dynamic-offer-driver-app/0127-updated-transporter-config-table.sql
Comment (umangj-28) [reply Backend/dev/migrations/dynamic-offer-driver-app/0127-updated-transporter-config-table.sql:1]: removed duplicate files

File: Backend/app/rider-platform/rider-app/Main/src/API/UI/Serviceability.hs [hunk 1]
```diff
@@ -15,15 +15,13 @@
 module API.UI.Serviceability
   ( API,
     handler,
-    ServiceabilityReq (..),
```
Comment (RitikaHanish) [inline Backend/app/rider-platform/rider-app/Main/src/API/UI/Serviceability.hs:18]: don't remove this

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Types/Serviceability.hs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{-# LANGUAGE UndecidableInstances #-}
```
Comment (RitikaHanish) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Types/Serviceability.hs]: remove this

**Review by RitikaHanish — APPROVED** [review]

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Types/Serviceability.hs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+module Domain.Types.Serviceability where
+
+import Kernel.External.Maps.Types
+import Kernel.Prelude
+import qualified Lib.Types.SpecialLocation as DSpecialLocation
+
+newtype ServiceabilityReq = ServiceabilityReq
```
Comment (hkmangla) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Types/Serviceability.hs:7]: These are not Domain types. Move them to Domain/Action only.

**Review by hkmangla — CHANGES_REQUESTED** [review]

**Review by hkmangla — APPROVED** [review]
